### PR TITLE
Add a task that prints the average training loss at each epoch.

### PR DESCRIPTION
**Pull request by MarcCote**
Because it's useful :p

### CODE REVIEW

**Comment by ASalvail** [comment]
I don't get it. Why is it needed versus the `Print` task? Furthermore, why isn't this a subclass of `Print`?

**Comment by MarcCote** [comment]
In order to compute the average training loss, we need the training loss at _each update_. That is, we need a special `Task` to keep a log of the losses (i.e. using the `post_update` method). Then, at the end of the epoch we take the average and want to print it.

The `Print` task prints one or multiple `View`s at a given updates/epochs frequency. However, a `View` is not a `Task`. A `View` uses lazy evaluation i.e. we only update its value when needed and it is only done once per update (we cache its results).

Right now, I don't see how to do it another way.

**Comment by ASalvail** [comment]
I'd rather we have a `Log` task that would take care of these kinds of situation. Actually, they might even replace the `View`s. 
What do you think?

**Comment by MarcCote** [comment]
This will be obsolete once PR #49 is merged.
